src/axl.c

```diff
@@ -490,9 +523,10 @@ static kvtree* AXL_Config_Get()
     /* this should be protected by a mutex_lock to prevent issues with
      * realloc() moving memory when growing axl_kvtrees, but no one else
      * does ... */
+    const int kvtrees_count = axl_kvtrees_count();
```

Comment:
You many need to hold the lock here before iterating over the all the kvtrees, as the count could change.

Reply:
Not sure if `count` matters. Since it only ever increases, I might be missing some transfers in the report (and it would only be changed from within the main thread so there is no race condition as long as only the main thread calls `AXL_Config`). The transfers themselves (`file_list`) are currently only changing from within the main thread. 

I have to admit I am weary of keeping the lock any longer than I need to and in particular calling any function while holding the lock (just in case those functions ever end up trying to get the lock as well and the mutex being of the lock-only-once type).

Reply:
uh, actually. I *do* need to hold the lock since the array pointer becomes invalid,l which i guess is what you had in mind.

Reply:
> uh, actually. I _do_ need to hold the lock since the array pointer becomes invalid,l which i guess is what you had in mind.

second uh. Maybe not in the end. As long as I am using the `avl_kvtree_get` function the change of pointer of the array does not affect me (and once more it should not change as long as I am on the main thread).

Reply:
The latest version no longer protects `axl_kvtrees` and instead copies the required entry (the `file_list` pointer) when a new transfer is created. This means this issue no longer applies.